### PR TITLE
Fix #21738 - Number of Stateful session beans in Passive state is not updated (always 0) in Jconsole

### LINKED ISSUE #21738: Number of Stateful session beans in Passive state is not updated (always 0) in Jconsole
<b>Issue:</b>
The number of Stateful session beans in Passive state (passivecount) is not updated at all in Jconsole. The value in Jconsole is always 0 even if session beans are passivated.
Make sure GLASSFISH-21272 has been fixed before reproducing this bug.

<b>Steps to Reproduce Issue:</b>
1. Set Monitoring Level for Ejb Container as HIGH.
2. Restart Glassfish.
3. Deploy test application and access the test application.
http://localhost:8080/StatefulBeansTest/MyServlet?count=600
=> Displays a "Hello" message in server.log, then starts passivating EJBs with a message for each.
4. Check value in Jconsole.
service:jmx:rmi://localhost:8686/jndi/rmi://localhost:8686/jmxrmi

amx > stateful-session-bean-mon > /mon/server-mon [server] > StatefulBeansEAR/StatefulBeansTest.war/PassivationBean > StatefulBeansEAR/StatefulEJB.jar/PassivationBean > Attributes

=> Click to expand the "passivecount" cell in the table
=> Bug: Should show non-zero highWaterMark (ie max) count but the current/low/high are all 0. When this bug is fixed, the current and highWaterMark become non-zero.
Test application:

I have used the test application from this website (EJB passivation and activation example).
https://www.javacodegeeks.com/2013/08/ejb-passivation-and-activation-example.html
I will attach the ear I built for your convenience.

### PR DESCRIPTION
**Pull request by srinivaskrishnan**
Fix #21738 - Number of Stateful session beans in Passive state is not updated (always 0) in Jconsole

### CODE REVIEW

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

File: main/appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/StatefulSessionContainer.java [hunk 1]
```diff
@@ -552,7 +552,13 @@ public String getMonitorAttributeValues() {
 
     protected EjbMonitoringStatsProvider getMonitoringStatsProvider(
             String appName, String modName, String ejbName) {
-        return new StatefulSessionBeanStatsProvider(this, getContainerId(), appName, modName, ejbName);
+        StatefulSessionBeanStatsProvider statsProvider = new StatefulSessionBeanStatsProvider(this, getContainerId(), appName, modName, ejbName);
+        try {
+            statsProvider.setPassiveCount(backingStore.size());
+        } catch (BackingStoreException e) {
+            _logger.log(Level.WARNING, "Exception in backingStore.size()", e);
```
Comment (yaminikb) [inline main/appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/StatefulSessionContainer.java]: Above message doesn't conform to GlassFish logging, should use LogMessageInfo

**Review by yaminikb — CHANGES_REQUESTED** [review]
Can you please add a test case?

**Comment by srinivaskrishnan** [comment]
With previous experience checking in the fix and adding a test case makes the hudson job/CI tests fail. So we need to merge the fix seperately and the test case seperately through different pull request.

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

**Comment by yaminikb** [comment]
Closing this PR since it is a duplicate.

**Pull request closed without merging**